### PR TITLE
[FEAT] Add countdown timer socket event on game start

### PR DESCRIPTION
**Pull request by dagea**
Added timer counting down from 3 on game start. Begins upon signalling of game start, and emits countdown information to frontend.

### CODE REVIEW

File: backend/src/domain/models/Game.js [hunk 1]
```diff
@@ -15,6 +15,31 @@ export class Game {
...
+   * Note: emits 3, 2, 1, 0.
+   * 0 represents the start of the game (calling nextRound)
+   */
+  startCountdown() {
+    this.io.to(this.session.sessionId).emit(
+        'countdown', {
+          count: this.countdown,
+        });
+
+    if (this.round === 0) {
```
Comment (afei088) [inline backend/src/domain/models/Game.js:32]: should this be this.countdown?

File: frontend/src/sockets/countdown.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Function for processing game start countdown
+ * @param {*} socket
+ * @param {*} cb call back function
+ */
+export function countdown(socket, cb) {
+  socket.on('countdown', (count) => {
```
Comment (afei088) [inline frontend/src/sockets/countdown.js:7]: an object is returned from the socket, so maybe use object destructuring? (e.g. ({count}) => {...})

Comment (dagea) [reply frontend/src/sockets/countdown.js]: ah true, will do

**Review by afei088 — APPROVED** [review]

**Review by David-Xia0 — DISMISSED** [review]

**Review by Tianrens — DISMISSED** [review]

**Review by Tianrens — APPROVED** [review]

**Comment by DavidHuangz** [comment]
Nice! Might want to hold off merging this for now as my countdown page is now called "CountDown." You can see it in
 https://github.com/we-are-number-1/yumble/pull/116

**Review by David-Xia0 — APPROVED** [review]

**Comment by dagea** [comment]
> Nice! Might want to hold off merging this for now as my countdown page is now called "CountDown." You can see it in
> #116

Hi, I don't think this should cause any issues? They're in different packages. I can rename if there are issues

**Comment by DavidHuangz** [comment]
> > Nice! Might want to hold off merging this for now as my countdown page is now called "CountDown." You can see it in
> > #116
> 
> Hi, I don't think this should cause any issues? They're in different packages. I can rename if there are issues

Okay, no problem, you can process through.

**Pull request merged**